### PR TITLE
Make exception handler accept Throwable superclasses

### PR DESCRIPTION
**Pull request by yurloc**

### CODE REVIEW

File: optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java [hunk 1]
```diff
@@ -157,6 +163,20 @@ public void exceptionInConsumer() throws InterruptedException {
...
     }
 
+    @Test
+    public void solveGenerics() {
+        final SolverConfig solverConfig = PlannerTestUtils
+                .buildSolverConfig(TestdataSolution.class, TestdataEntity.class);
+        SolverManager<TestdataSolution, Long> solverManager = SolverManager
+                .create(solverConfig, new SolverManagerConfig());
+
+        BiConsumer<Object, Object> exceptionHandler = (o1, o2) -> fail("This is unexpected");
```
Comment (yurloc) [inline optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java]: @ge0ffrey This type of exception handler couldn't be used with the `solve()` method.

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java [hunk 2]
```diff
@@ -40,16 +40,17 @@
...
+    private final BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler;
 
     private volatile SolverStatus solverStatus;
     private Future<Solution_> future;
 
-    public DefaultSolverJob(DefaultSolverManager<Solution_, ProblemId_> solverManager, ProblemId_ problemId,
-            Solver<Solution_> solver,
+    public DefaultSolverJob(
+            DefaultSolverManager<Solution_, ProblemId_> solverManager,
+            Solver<Solution_> solver, ProblemId_ problemId,
```
Comment (yurloc) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java]: Changed order of parameters (solverManager + solver first, then problemId + problemFinder).

**Comment by yurloc** [comment]
@ge0ffrey ping

**Comment by ge0ffrey** [comment]
Sorry for the delay, merged :)

File: optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java [hunk 3]
```diff
@@ -157,6 +163,20 @@ public void exceptionInConsumer() throws InterruptedException {
...
+        Consumer<Object> finalBestSolutionConsumer = o -> {};
+        Function<Object, TestdataUnannotatedExtendedSolution> problemFinder = o -> new TestdataUnannotatedExtendedSolution();
+
+        solverManager.solve(1L, problemFinder, finalBestSolutionConsumer, exceptionHandler);
```
Comment (ge0ffrey) [inline optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java]: This solver actually failed, but that failure never reached the test thread. I believe this change fixed it, please review :)
https://github.com/kiegroup/optaplanner/commit/d2589ed10073618ec7cbea7bfe0ce6cac55393ae

**Pull request merged**